pygmt/tests/test_surface.py

```diff
@@ -114,12 +114,13 @@ def test_surface_short_aliases(ship_data):
     outgrid.
     """
     data = ship_data.values  # convert pandas.DataFrame to numpy.ndarray
-    try:
-        output = surface(data=data, I="5m", R=[245, 255, 20, 30], G=TEMP_GRID)
-        assert output is None  # check that output is None since outgrid is set
-        assert os.path.exists(path=TEMP_GRID)  # check that outgrid exists at path
-        with xr.open_dataarray(TEMP_GRID) as grid:
-            assert isinstance(grid, xr.DataArray)  # ensure netcdf grid loads ok
-    finally:
-        os.remove(path=TEMP_GRID)
-    return output
+    with pytest.warns(expected_warning=SyntaxWarning) as record:
+        try:
+            output = surface(data=data, I="5m", R=[245, 255, 20, 30], G=TEMP_GRID)
+            assert output is None  # check that output is None since outgrid is set
+            assert os.path.exists(path=TEMP_GRID)  # check that outgrid exists at path
+            with xr.open_dataarray(TEMP_GRID) as grid:
+                assert isinstance(grid, xr.DataArray)  # ensure netcdf grid loads ok
+        finally:
+            os.remove(path=TEMP_GRID)
```

Comment:
Maybe we should refactor the test to use GMTTempFile instead of try-except?

Reply:
Good idea. I was just trying to suppress the warnings here, so that I wouldn't need to scroll so much to inspect failures in another branch. But if you want, I could expand this to address your comment and https://github.com/GenericMappingTools/pygmt/issues/1460 in this PR.

Reply:
Yep, I'm ok with a big refactor PR.

Reply:
OK, I have refactored the tests in this PR. I'll mark this conversation as resolved and we can discuss any necessary changes in a separate review.